sotodlib/io/g3tsmurf_utils.py

```diff
@@ -0,0 +1,195 @@
+        as  `load_file(... samples = list[i] ... )`
+    test: bool
+        If true, yields a tuple of (det_chunks, samp_chunks) instead of a loaded AxisManager
+    load_file_kwargs: dict
+        additional arguments to pass to load_smurf
+    
+    Yields
+    --------
+    AxisManagers with loaded sections of data
+    """
+    
+    session = archive.Session()
+    obs = session.query(Observations).filter(Observations.obs_id==obs_id).one()
+    db_files = session.query(Files).filter(Files.obs_id==obs_id).order_by(Files.start)
+    filenames = sorted( [f.name for f in db_files])
+    
+    ts = obs.tunesets[0] ## if this throws an error we have some fallbacks
+    obs_dets, obs_samps = len(ts.dets), obs.n_samples
+    
+    logger.debug(f"{obs_id} has (n_dets, n_samps): ({obs_dets}, {obs_samps})")
+    if ram_limit is not None:
+        pts_limit = int(ram_limit // 4)
+        n_samp_chunks = 1
+        n_samps = obs_samps
+        n_dets = pts_limit // n_samps   
+        while n_dets == 0:
+            n_samp_chunks += 1
+            n_samps = obs_samps//n_samp_chunks
+            n_dets= pts_limit // (n_samps)
+        n_det_chunks = int(np.ceil( obs_dets/n_dets ))
+
+    if n_det_chunks is not None and n_dets is not None:
+        logger.warning("Both n_det_chunks and n_dets specified, n_det_chunks overrides")
+    if n_samp_chunks is not None and n_samps is not None:
+        logger.warning("Both n_samp_chunks and n_samps specified, n_samp_chunks overrides")
+        
+    if n_det_chunks is not None:
+        n_dets = int(np.ceil(obs_dets/n_det_chunks))
+        det_chunks = [range(i*n_dets,min((i+1)*n_dets,obs_dets)) for i in range(n_det_chunks)]
+    if n_samp_chunks is not None:
+        n_samps = int(np.ceil(obs_samps/n_samp_chunks))
+        samp_chunks = [(i*n_samps,min((i+1)*n_samps,obs_samps)) for i in range(n_samp_chunks)]
+
+    if n_dets is not None:
+        n_det_chunks = int(np.ceil(obs_dets/n_dets))
+        det_chunks = [range(i*n_dets,min((i+1)*n_dets,obs_dets)) for i in range(n_det_chunks)]
+    if n_samps is not None:
+        n_samp_chunks = int(np.ceil(obs_samps/n_samps))
+        samp_chunks = [(i*n_samps,min((i+1)*n_samps,obs_samps)) for i in range(n_samp_chunks)]
+        
+    if det_chunks is None:
+        det_chunks = [range(0,obs_dets)]
+    if samp_chunks is None:
+        samp_chunks = [(0,obs_samps)]
+        
+    ## should we let folks overwrite this here?
+    if "archive" in load_file_args:
+        archive = load_file_args.pop("archive")
+        
+    logger.debug(f"Loading data with det_chunks: {det_chunks}.")
+    logger.debug(f"Loading data in samp_chunks: {samp_chunks}.")
+    
+    try:
+        for det_chunk in det_chunks:
+            for samp_chunk in samp_chunks:
+                if test:
+                    yield (det_chunk, samp_chunk)
+                else:
+                    yield load_file( 
```

Comment:
How long does it take for the SmurfStatus to be loaded? Maybe that should be loaded once and passed in to each batch.

Reply:
not long but it is additional IO, I'll add it this.